integ-tests/src/main/java/com/google/pojo/UserTableRow.java

```diff
@@ -25,29 +27,58 @@
 
   @Override
   public String toString() {
-    return "UserTableRow: {"
-        + "userName: {"
+    return "{"
+        + "UserName: {"
         + userName.toString()
         + "}"
-        + ", creatorName: {"
+        + ", CreatorName: {"
         + creatorName.toString()
         + "}"
-        + ", createTimeStamp: {"
+        + ", CreateTimeStamp: {"
         + createTimeStamp.toString()
         + "}"
-        + ", lastAccessTimeStamp: {"
+        + ", LastAccessTimeStamp: {"
         + lastAccessTimeStamp.toString()
         + "}"
         + '}';
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof UserTableRow)) return false;
+    UserTableRow that = (UserTableRow) o;
+    return Objects.equals(userName, that.userName)
+        && Objects.equals(creatorName, that.creatorName)
+        && Objects.equals(createTimeStamp, that.createTimeStamp)
+        && Objects.equals(lastAccessTimeStamp, that.lastAccessTimeStamp);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(userName, creatorName, createTimeStamp, lastAccessTimeStamp);
+  }
+
   /** Inner POJO class for serialization data from DB and Avro files. */
   public class UserName {
```

Comment:
I don't mind doing this but what's the reason? 
Could you briefly explain?
I thought then less things you keep static then better from a memory utilization perspective.

Reply:
By having the class be non-static, the instance needs to keep a reference to the enclosing class instance. It can also only be instantiated from within an instance of the outer class.

Why would having an inner class be non-static be better for memory utilization?

Reply:
Done.